migrations/2023-07-23-182318_site_aggregates_local_only/up.sql

```diff
@@ -0,0 +1,130 @@
+-- to ensure no confusion with all the migrations
+--    this was mostly created by dumping from PostgreSQL 15.3 schema
+-- note: hard-coded site_id = 1 is etablished convention in Lemmy's Rust code for local site.
+
+create unique index idx_site_aggregates_site_id on site_aggregates (site_id);
+
+CREATE OR REPLACE FUNCTION  site_aggregates_comment_delete() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+begin
+    IF (was_removed_or_deleted(TG_OP, OLD, NEW)) THEN
+        update site_aggregates
+        set comments = comments - 1
+        where site_id = 1;
```

Comment:
If you look at the triggers, they're already only firing when the content is local:

`site_aggregates_comment_delete AFTER DELETE OR UPDATE OF removed, deleted ON comment FOR EACH ROW WHEN (old.local = true) EXECUTE FUNCTION site_aggregates_comment_delete()`

So I don't see that this would do anything. 

Rather than adding this check, we should delete the site_aggregate rows for non-local sites, and see if there's any other place they *might* get inserted. The triggers, rather than the functions, are the ideal place to catch this.

Reply:
Then the issue is the site_aggregates_site trigger, which inserts a row into site_aggregates for every insert into site. What should that filter though by? By the existence of local_site with same site_id?

Reply:
The local site always has id = 1 so the trigger only needs to run for that.